File: app/src/main/java/com/script/CompiledScript.java
/*
 * Decompiled with CFR 0.152.
 */
package com.script;

public abstract class CompiledScript {
    public abstract Object eval(ScriptContext var1) throws ScriptException;

    public abstract ScriptEngine getEngine();

    public Object eval(Bindings bindings) throws ScriptException {
        ScriptContext ctxt = this.getEngine().getContext();
        if (bindings != null) {
            SimpleScriptContext tempctxt = new SimpleScriptContext();
            tempctxt.setBindings(bindings, 100);
            tempctxt.setBindings(ctxt.getBindings(200), 200);
            tempctxt.setWriter(ctxt.getWriter());
            tempctxt.setReader(ctxt.getReader());
            tempctxt.setErrorWriter(ctxt.getErrorWriter());
            ctxt = tempctxt;
        }
        return this.eval(ctxt);
    }

    public Object eval() throws ScriptException {
        return this.eval(this.getEngine().getContext());
    }
}

File: modules/rhino1.7.3/src/main/java/com/script/CompiledScript.kt
/*
 * Decompiled with CFR 0.152.
 */
package com.script

import org.mozilla.javascript.Scriptable
import kotlin.coroutines.CoroutineContext

abstract class CompiledScript {

    abstract fun getEngine(): ScriptEngine

    @Throws(ScriptException::class)
    abstract fun eval(context: ScriptContext): Any?

    @Throws(ScriptException::class)
    abstract fun eval(scope: Scriptable): Any?

    @Throws(ScriptException::class)
    abstract fun eval(scope: Scriptable, coroutineContext: CoroutineContext?): Any?

    @Throws(ScriptException::class)
    abstract suspend fun evalSuspend(scope: Scriptable): Any?

    @Throws(ScriptException::class)
    fun eval(bindings: Bindings?): Any? {
        var ctxt = getEngine().context
        if (bindings != null) {
            val tempContext = SimpleScriptContext()
            tempContext.setBindings(bindings, 100)
            tempContext.setBindings(ctxt.getBindings(200), 200)
            tempContext.writer = ctxt.writer
            tempContext.reader = ctxt.reader
            tempContext.errorWriter = ctxt.errorWriter
            ctxt = tempContext
        }
        return this.eval(ctxt)
    }

    @Throws(ScriptException::class)
    fun eval(): Any? {
        return this.eval(getEngine().context)
    }
}